src/components/autonomous/WaypointNav.js

```diff
@@ -0,0 +1,73 @@
+import { useDispatch, useSelector } from "react-redux";
+import { selectLatitude, selectLongitude, selectIsApproximate, selectIsGated, requestWaypointNav } from "../../store/waypointNavSlice";
+import { selectOpMode } from "../../store/opModeSlice";
+import "./WaypointNav.css";
+import { useEffect } from "react";
+
+function WaypointNav() {
+  const dispatch = useDispatch();
+
+  var latitudeGiven = useSelector(selectLatitude);
+  var longitudeGiven = useSelector(selectLongitude);
+  var isApproximate = useSelector(selectIsApproximate);
+  var isGated = useSelector(selectIsGated);
+  var opMode = useSelector(selectOpMode);
+
+  const handleClick = () => {
+    console.log("hey");
+    updateLatitude();
+    updateLongitude();
+    updateIsApproximate();
+    updateIsGate();
+
+    // Only send the event in autonomous mode
+    if (opMode === "autonomous") {
+        dispatch(requestWaypointNav({ latitude: latitudeGiven,
+                                      longitude: longitudeGiven,
+                                      approximate: isApproximate,
+                                      gated: isGated}));
+    }
+
+  };
+
+  const updateLatitude = () => {
+    latitudeGiven = document.getElementById('latitude').value;
+  };
+
+  const updateLongitude = () => {
+    longitudeGiven = document.getElementById('longitude').value;
+  };
+
+  const updateIsApproximate = () => {
+    isApproximate = document.getElementById('isApproximate').checked;
+  };
+
+
+  const updateIsGate = () => {
+    isGated = document.getElementById('isGate').checked;
+  };
```

Comment:
All of these methods should be deleted. Since we're in react we don't want to manually interact with the DOM